shotover/src/transforms/tee.rs

```diff
@@ -242,12 +322,115 @@ impl Transform for Tee {
                             .collect::<Vec<_>>()
                     );
                 }
-                Ok(chain_response)
+                self.return_response(tee_response, chain_response).await
             }
         }
     }
 }
 
+struct ChainSwitchListener {
+    address: SocketAddr,
+}
+
+impl ChainSwitchListener {
+    fn new(address: SocketAddr) -> Self {
+        Self { address }
+    }
+
+    fn rsp(status: StatusCode, body: impl Into<Body>) -> Response<Body> {
+        Response::builder()
+            .status(status)
+            .body(body.into())
+            .expect("builder with known status code must not fail")
+    }
+
+    async fn set_result_source_chain(
+        body: Bytes,
+        result_source: Arc<AtomicResultSource>,
+    ) -> Result<()> {
+        let new_result_source = str::from_utf8(body.as_ref())?;
+
+        let new_value = match new_result_source {
+            "tee-chain" => ResultSource::TeeChain,
+            "regular-chain" => ResultSource::RegularChain,
+            _ => {
+                return Err(anyhow!(
+                    r"Invalid value for result source: {}, should be 'tee-chain' or 'regular-chain'",
+                    new_result_source
+                ))
+            }
+        };
+
+        debug!("Setting result source to {}", new_value);
+
+        result_source.store(new_value, Ordering::Relaxed);
+        Ok(())
+    }
+
+    async fn async_run(self, result_source: Arc<AtomicResultSource>) {
+        if let Err(err) = self.async_run_inner(result_source).await {
+            error!("Error in ChainSwitchListener: {}", err);
+        }
+    }
+
+    async fn async_run_inner(self, result_source: Arc<AtomicResultSource>) -> Result<()> {
+        let make_svc = make_service_fn(move |_| {
+            let result_source = result_source.clone();
+            async move {
+                Ok::<_, Infallible>(service_fn(move |req: Request<Body>| {
+                    let result_source = result_source.clone();
+                    async move {
+                        let response = match (req.method(), req.uri().path()) {
+                            (&Method::GET, "/result-source") => {
```

Comment:
Its quite likely that we will want to combine this webservice with our metrics webservice.
So I wonder if we should namespace this like `/transform/tee/result-source` in an attempt to make the transition to such a setup just a matter of changing ports.
Trying to guess what exactly we will need the API to be is futile but this is at least a bit closer to what we will need.

Reply:
I thought about it more, lets just leave it as is, we'll need some kind of naming system to address which tee we actually want to configure.